src/transports/ssh.c

```diff
@@ -893,9 +913,24 @@ int git_transport_ssh_with_paths(git_transport **out, git_remote *owner, void *p
 #endif
 }
 
+void git_transport_ssh_global_shutdown(void);
+
 int git_transport_ssh_global_init(void)
 {
 #ifdef GIT_SSH
+#ifdef GIT_WIN32
+	const char *(*libssh2_version)(int) = NULL;
+#else
+	libssh2_handle = dlopen("libssh2.dylib", RTLD_LAZY);
+	const char *(*libssh2_version)(int) = dlsym(libssh2_handle, "libssh2_version");
+#endif
+
+	// Check that libssh2's version is sufficient for us
+	if (!libssh2_version || !libssh2_version(LIBSSH2_VERSION_NUM)) {
+		return 0;
+	}
```

Comment:
I do agree it's kinda unclear, but if we were to report an error here, that would make the library completely fail to initialize. Right now this stashes the handle away, and the actual error-returning check is done when we attempt to create a transport — mostly because handles are ref-counted, and one handle per library initialization Ought To Be Enough For Anybody.